manifests/quest.pp

```diff
@@ -1,12 +1,66 @@
 class learning::quest ($git_branch='release') {
   
-  $doc_root = '/var/www/html/questguide'
-  $port     = '80'
+  $doc_root = '/var/www/html/questguide/'
+  $proxy_port = '80'
+  $quest_port = '85'
+  $graph_port = '90'
 
-  class { 'apache':
+  class { '::apache':
+    default_vhost => false,
+  }
+
+  # Create a proxy to pass to the Quest Guide and graphite server
+
+  apache::vhost { "*:${proxy_port}":
+    port       => $proxy_port,
+    docroot    => $doc_root,
+    proxy_pass => [
+      { 'path' => '/graphite', 'url' => "http://localhost:${graph_port}"},
+      { 'path' => '*', 'url' => "http://localhost:${quest_port}"},
+    ],
+  }
+
+  # Serve the Quest Guide
+
+  apache::vhost { "*:${quest_port}":
+    port    => $quest_port,
     docroot => $doc_root,
   }
 
+  # Serve Graphite
```

Comment:
Would we want to ensure that graphite is installed, configured and that /opt/graphite/<directories> exists etc here? Or, what happens if those directories are absent? What happens if the graphite conf changes, would apache (or its conf) need to be restarted (or reloaded)? Just checking.